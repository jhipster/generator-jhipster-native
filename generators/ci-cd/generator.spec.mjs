import expect from 'expect';

import { helpers, lookups } from '#test-utils';

describe('SubGenerator ci-cd of native JHipster blueprint', () => {
  describe('run', () => {
    let result;
    before(async function () {
      result = await helpers
        .create('jhipster:ci-cd')
        .withOptions({
          reproducible: true,
          defaults: true,
          blueprint: 'native',
          autoconfigureGithub: true,
          localConfig: {
            baseName: 'jhipster',
            testFrameworks: ['cypress'],
          },
        })
        .withLookups(lookups)
        .run();
    });

    it('should match snapshot', () => {
      expect(result.getSnapshot('**/.github/workflows/native.yml')).toMatchSnapshot();
    });
  });
});
