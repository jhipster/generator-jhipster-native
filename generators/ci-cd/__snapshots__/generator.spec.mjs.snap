// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubGenerator ci-cd of native JHipster blueprint run should match snapshot 1`] = `
Object {
  ".github/workflows/native.yml": Object {
    "contents": "name: Native CI
on:
  workflow_dispatch:
  schedule:
    - cron: \\"0 0 * * *\\"
jobs:
  pipeline:
    name: jhipster native pipeline
    runs-on: ubuntu-latest
    if: \\"!contains(github.event.head_commit.message, '[ci skip]') && !contains(github.event.head_commit.message, '[skip ci]') && !contains(github.event.pull_request.title, '[skip ci]') && !contains(github.event.pull_request.title, '[ci skip]')\\"
    timeout-minutes: 40
    env:
      SPRING_OUTPUT_ANSI_ENABLED: DETECT
      SPRING_JPA_SHOW_SQL: false
      JHI_DISABLE_WEBPACK_LOGS: true
      NG_CLI_ANALYTICS: false
    steps:
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
        with:
          node-version: 16.14.0
      - name: \\"SETUP: setup graalvm\\"
        uses: graalvm/setup-graalvm@v1
        with:
          version: \\"latest\\"
          java-version: \\"11\\"
          components: \\"native-image\\"
          github-token: \${{ secrets.GITHUB_TOKEN }}
      - name: Install node.js packages
        run: npm install
      - name: \\"E2E: Package\\"
        run: npm run native-package
      - name: \\"E2E: Prepare\\"
        run: npm run ci:e2e:prepare
      - name: \\"E2E: Run\\"
        run: npm run native-e2e
",
    "stateCleared": "modified",
  },
}
`;
